File: framework/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/model/ProductListerWidget.java
package com.sdl.ecommerce.dxa.model;

import com.sdl.ecommerce.api.Query;
import com.sdl.ecommerce.api.QueryInputContributor;
import com.sdl.ecommerce.api.model.Product;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;

import java.util.List;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * Product Lister Widget
 *
 * @author nic
 */
@SemanticEntity(entityName = "ProductListerWidget", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class ProductListerWidget extends AbstractEntityModel implements QueryInputContributor {

    @SemanticProperty("e:category")
    private ECommerceCategoryReference categoryReference;

    @SemanticProperty("e:viewSize")
    private int viewSize = 0;

    @SemanticProperty("e:viewType")
    private String viewType;

    // TODO: Add a semantic property
    private Boolean disableRedirect = null;

    private List<Product> items;

    private String nextUrl;
    private String previousUrl;
    private String lastUrl;
    private String firstUrl;
    private boolean showNavigation = false;
    private int currentSet;
    private int viewSets;

    @Override
    public void contributeToQuery(Query query) {
        if ( viewSize > 0 ) {
            query.viewSize(viewSize);
        }
        // TODO: HOW TO HANDLE FH SPECIFIC CONTRIBUTIONS?????
        /*
        if ( disableRedirect != null ) {
            queryConfiguration.setDisableRedirect(disableRedirect);
        } */
    }

    public ECommerceCategoryReference getCategoryReference() {
        return categoryReference;
    }

    public int getViewSize() {
        return viewSize;
    }

    public String getViewType() {
        return viewType;
    }

    public List<Product> getItems() {
        return items;
    }

    public void setItems(List<Product> items) {
        this.items = items;
    }

    public String getNextUrl() {
        return nextUrl;
    }

    public void setNextUrl(String nextUrl) {
        this.nextUrl = nextUrl;
    }

    public String getPreviousUrl() {
        return previousUrl;
    }

    public void setPreviousUrl(String previousUrl) {
        this.previousUrl = previousUrl;
    }

    public String getLastUrl() {
        return lastUrl;
    }

    public void setLastUrl(String lastUrl) {
        this.lastUrl = lastUrl;
    }

    public String getFirstUrl() {
        return firstUrl;
    }

    public void setFirstUrl(String firstUrl) {
        this.firstUrl = firstUrl;
    }

    public boolean isShowNavigation() {
        return showNavigation;
    }

    public void setShowNavigation(boolean showNavigation) {
        this.showNavigation = showNavigation;
    }

    public int getCurrentSet() {
        return currentSet;
    }

    public void setCurrentSet(int currentSet) {
        this.currentSet = currentSet;
    }

    public int getViewSets() {
        return viewSets;
    }

    public void setViewSets(int viewSets) {
        this.viewSets = viewSets;
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/ProductListerWidget.cs
﻿using Sdl.Web.Common.Models;
using SDL.ECommerce.Api.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.DXA.Models
{
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "ProductListerWidget", Prefix = "e")]
    public class ProductListerWidget : EntityModel, IQueryContributor
    {
        [SemanticProperty("e:category")]
        public ECommerceCategoryReference CategoryReference { get; set; }

        [SemanticProperty("e:viewSize")]
        public int? ViewSize { get; set; }

        [SemanticProperty("e:viewType")]
        public string ViewType { get; set; }

        [SemanticProperty(IgnoreMapping = true)]
        public IList<IProduct> Items { get; set; }
    
        [SemanticProperty(IgnoreMapping = true)]
        public ListerNavigationData NavigationData { get; set; }

        public void ContributeToQuery(Api.Model.Query query)
        {
            if ( ViewSize != null )
            {
                query.ViewSize = ViewSize;
            }
        }
    }
}
